s3_management/manage.py

```diff
@@ -212,6 +213,23 @@ def normalize_package_version(self: S3IndexType, obj: str) -> str:
     def obj_to_package_name(self, obj: str) -> str:
         return path.basename(obj).split('-', 1)[0]
 
+    def fetch_checksum_from_s3(self, s3_key):
+        s3_key = s3_key.replace("%2B", "+")
+        try:
+            response = CLIENT.get_object_attributes(
+                Bucket=BUCKET,
+                Key=s3_key,
+                ObjectAttributes=['Checksum']
+            )
+            checksum = response['Checksum']['ChecksumSHA256']
+            return checksum
+        except NoCredentialsError:
+            print("No AWS credentials found")
+            return None
+        except Exception as e:
```

Comment:
This is a wide exception. I think the `try:` block and relevant exceptions likely should be scoped to be explicit about what is expected to fail and what isn't.

Otherwise, a simple bug could manifest as an index not having hashes. When the alternative is an error with a stacktrace given to the user, and no index generated.

Reply:
Resolved.